### PR TITLE
UX-717 Adds new z-index token, Fix textfield z-indexing with error outlines

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Fixes the right border of the textfield with right-connected components

### How To Test or Verify
- Visit: http://localhost:9001/?path=Visual-Regression__TextField
- Test focus styles on the last textfield with an error

Screen Shot 2021-12-16 at 8 53 30 AM

Screen Shot 2021-12-16 at 8 53 38 AM

### PR Checklist

- [ ] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/Connect/Connect.tsx [hunk 1]
```diff
@@ -3,10 +3,13 @@ import { Box } from '../Box';
...
 
-const FocusHandler = styled(Box)`
+// $zIncrement is here to ensure the middle element appears
+// above side elements until it side elements are focused
+const FocusHandler = styled(Box)<{ $zIncrement: number }>`
+  z-index: ${(props) => tokens.zIndex_default + props.$zIncrement};
   &:focus-within {
     position: relative;
-    z-index: ${tokens.zIndex_default};
+    z-index: ${tokens.zIndex_default + 2};
```
Comment (logansparlin) [inline packages/matchbox/src/components/Connect/Connect.tsx:12]: This works but seems a little odd, maybe we need to consider a new zIndex token?

**Review by logansparlin — APPROVED** [review]
Looks good! Just one small comment

File: packages/matchbox/src/components/Connect/Connect.tsx (hunk 1 above)
Comment (jonambas) [reply packages/matchbox/src/components/Connect/Connect.tsx:12]: would `above` work? with a value of 10 just to be safe?

Comment (logansparlin) [reply packages/matchbox/src/components/Connect/Connect.tsx:12]: Yea let's do that

**Pull request merged**